File: debugging/src/main/java/com/barista_v/debugging/item/spinner/SpinnerDrawerItem.java
package com.barista_v.debugging.item.spinner;

import android.support.v7.widget.PopupMenu;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import com.barista_v.debugging.R;
import com.mikepenz.materialdrawer.model.PrimaryDrawerItem;
import java.util.ArrayList;
import java.util.List;

public class SpinnerDrawerItem extends PrimaryDrawerItem implements View.OnClickListener,
    PopupMenu.OnMenuItemClickListener {

  final String[] mItems;
  final List<SpinnerItemListener> mListeners;

  public SpinnerDrawerItem(int id, String[] items, final SpinnerItemListener spinnerItemListener,
      String selectedItem) {
    this(id, items, spinnerItemListener, java.util.Arrays.asList(items).indexOf(selectedItem));
  }

  public SpinnerDrawerItem(int id, String[] items, final SpinnerItemListener spinnerItemListener,
      int selectedItemIndex) {
    mItems = items;
    mListeners = new ArrayList<SpinnerItemListener>() {{
      add(spinnerItemListener);
    }};

    withTag(id);
    withDescription(mItems[selectedItemIndex]);
    withIcon(R.drawable.ic_arrow_drop_down_grey_700_24dp);
  }

  @Override
  public void bindView(ViewHolder viewHolder) {
    super.bindView(viewHolder);
    viewHolder.itemView.setOnClickListener(this);
  }

  //<editor-fold desc="OnClickListener">
  @Override
  public void onClick(View v) {
    PopupMenu popupMenu = new PopupMenu(v.getContext(), v);
    Menu menu = popupMenu.getMenu();

    for (int i = 0, length = mItems.length; i < length; i++) {
      menu.add(mItems[i]);
    }

    popupMenu.setOnMenuItemClickListener(this);
    popupMenu.show();
  }
  //</editor-fold>

  //<editor-fold desc="OnMenuItemClickListener">
  @Override
  public boolean onMenuItemClick(MenuItem item) {
    for (int i = 0, length = mListeners.size(); i < length; i++) {
      mListeners.get(i).onSpinnerItemClick(this, (int) getTag(), item.getTitle());
    }
    return true;
  }
  //</editor-fold>

  public SpinnerDrawerItem withMoreListeners(SpinnerItemListener listener) {
    mListeners.add(listener);
    return this;
  }
}

File: lib/src/main/java/debug_artist/menu/drawer/item/spinner/SpinnerDrawerItem.kt
package debug_artist.menu.drawer.item.spinner

import android.support.v7.widget.PopupMenu
import android.view.MenuItem
import android.view.View
import com.mikepenz.materialdrawer.model.PrimaryDrawerItem
import debug_artist.menu.R

open class SpinnerDrawerItem(id: Int,
                             internal val items: Array<String>,
                             spinnerItemListener: SpinnerItemListener,
                             selectedItemIndex: Int) :
    PrimaryDrawerItem(), View.OnClickListener, PopupMenu.OnMenuItemClickListener {

  internal val listeners = mutableListOf(spinnerItemListener)

  init {
    withTag(id)

    if (selectedItemIndex != -1) {
      withDescription(items[selectedItemIndex])
    }

    withIcon(R.drawable.ic_arrow_drop_down_grey_700_24dp)
  }

  override fun bindView(viewHolder: ViewHolder?, payloads: MutableList<Any?>) {
    super.bindView(viewHolder, payloads)
    viewHolder?.itemView?.setOnClickListener(this)
  }

  //<editor-fold desc="OnClickListener">
  override fun onClick(v: View) {
    PopupMenu(v.context, v).apply {
      items.forEach { menu.add(it) }
      setOnMenuItemClickListener(this@SpinnerDrawerItem)
      show()
    }
  }
  //</editor-fold>

  //<editor-fold desc="OnMenuItemClickListener">
  override fun onMenuItemClick(item: MenuItem): Boolean {
    listeners.forEach { it.onSpinnerItemClick(this, tag as Int, item.title) }
    return true
  }
  //</editor-fold>

  fun withMoreListeners(listener: SpinnerItemListener): SpinnerDrawerItem {
    listeners.add(listener)
    return this
  }
}
